py_gtfs_rt_ingestion/ingest.py

```diff
@@ -9,11 +9,121 @@
 
 from py_gtfs_rt_ingestion import Configuration
 
+
 DESCRIPTION = "Convert a json file into a parquet file. Used for testing."
+MULTIPROCESSING_POOL_SIZE = 4
+
+def gz_to_pyarrow(filename: str, config: Configuration, return_queue: Queue) -> None:
+    """
+    Accepts filename as string. Converts gzipped json -> pyarrow table. 
+
+    parrow table sent back to main process using return_queue.
+    If conversion fails, returns filename.
+    """
+    # Enclose entire function in try block, Process can't fail silently.
+    # Must return either pyarrow table or filename
+    try:
+        with gzip.open(filename, 'rb') as f:
+            json_data = json.loads(f.read())
+
+        # parse timestamp info out of the header
+        header = json_data['header']
+        feed_timestamp = header['timestamp']
+        timestamp = datetime.utcfromtimestamp(feed_timestamp)
+
+        # create dict of lists 
+        ret_obj = {key.name:[] for key in config.export_schema}
+
+        # for each entity in the list, create a record and add it to the table
+        for entity in json_data['entity']:
+            record = config.record_from_entity(entity=entity)
+            record.update({
+                'year': timestamp.year,
+                'month': timestamp.month,
+                'day': timestamp.day,
+                'hour': timestamp.hour,
+                'feed_timestamp': feed_timestamp
+            })
+
+            for key in record:
+                ret_obj[key].append(record[key])
+
+        ret_obj = pa.table(ret_obj, schema=config.export_schema)
+
+    # Catch failure, send back filename for handling
+    except Exception as e:
+        print(e, flush=True)
+        ret_obj = filename
+
+    # Send pyarrow table or filename back to main Process for concatenation
+    return_queue.put(ret_obj)
+    return_queue.close()
+
+def lambda_handler(event: dict, context) -> None:
+    """
+    AWS Lambda Python handled function as described in AWS Developer Guide:
+    https://docs.aws.amazon.com/lambda/latest/dg/python-handler.html
+    :param event: The event dict sent by Amazon API Gateway that contains all of the
+                  request data.
+    :param context: The context in which the function is called.
+    :return: A response that is sent to Amazon API Gateway, to be wrapped into
+             an HTTP response. The 'statusCode' field is the HTTP status code
+             and the 'body' field is the body of the response.
+
+    expected event structure:
+    {
+        file_batch: [file_name_1, file_name_2, ...],
+    }
+    batch files should all be of same ConfigType
+    """
+    if 'file_batch' not in event:
+        raise Exception("Lambda event missing 'file_batch' key.")
+
+    file_batch = event['file_batch']
+
+    config = Configuration(file_batch[0])
+
+    return_queue = Queue()
+
+    pool = Pool(MULTIPROCESSING_POOL_SIZE)
+    pool.starmap_async(gz_to_pyarrow, [(f, config, return_queue) for f in file_batch])
```

Comment:
Set up multiprocessing, create Queue for return data and Pool object. 

Call `starmap_async` so process work does not block this process collecting data from `return_queue`.